client/dashboard/sites/overview/visibility-card.tsx

```diff
@@ -0,0 +1,112 @@
+import { CircularProgressBar } from '@automattic/components';
+import { __experimentalText as Text } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { lockOutline, published } from '@wordpress/icons';
+import { launch } from '../../components/icons';
+import { isSelfHostedJetpackConnected } from '../../utils/site-types';
+import OverviewCard from '../overview-card';
+import type { Site } from '../../data/types';
+
+const CARD_PROPS = {
+	title: __( 'Visibility' ),
+	trackId: 'visibility',
+};
+
+function getVisibilityURL( site: Site ) {
+	if ( isSelfHostedJetpackConnected( site ) ) {
+		return undefined;
+	}
+
+	return `/sites/${ site.slug }/settings/site-visibility`;
+}
+
+function VisibilityCardUnlaunched() {
+	const isSetupComplete = true;
+	let heading = __( 'Coming soon' );
+	let description = __( 'Finish setting up your site' );
+
+	if ( isSetupComplete ) {
+		heading = __( 'Launch site' );
+		description = __( 'Ready to go public?' );
+	}
+
+	return (
+		<OverviewCard
+			{ ...CARD_PROPS }
+			icon={ launch }
+			heading={ heading }
+			description={ description }
+			sideContent={
+				<CircularProgressBar
+					currentStep={ 5 }
+					numberOfSteps={ 5 }
+					size={ 80 }
+					strokeWidth={ 1.5 }
+					variant="success"
+					customText={
+						<Text lineHeight="20px" size={ 15 } weight={ 500 }>
+							5/5
+						</Text>
+					}
+				/>
+			}
+		/>
+	);
+}
+
+function VisibilityCardComingSoon( { site }: { site: Site } ) {
+	return (
+		<OverviewCard
+			{ ...CARD_PROPS }
+			icon={ launch }
+			heading={ __( 'Coming soon' ) }
+			description={ __( 'Ready to go public?' ) }
+			link={ getVisibilityURL( site ) }
+		/>
+	);
+}
+
+function VisibilityCardPrivate( { site }: { site: Site } ) {
+	return (
+		<OverviewCard
+			{ ...CARD_PROPS }
+			icon={ lockOutline }
+			heading={ __( 'Private' ) }
+			description={ __( 'Only invited users can view your site' ) }
```

Comment:
We always want to end sub-headings/descriptions with periods right @matt-west?

Since multi-sentence paragraphs need periods, to prevent special casing, we want single sentences to have periods too rykv85Ii9RkvP504dtqqaW-fi-159_11853